src/main-client.ts

```diff
@@ -1145,6 +1158,67 @@ export class MainClient extends BaseRestClient {
     );
   }
 
+  /**
+   *
+   * Simple earn Endpoints
+   *
+   **/
+
+  getFlexibleSavingProducts(
+    params: SimpleEarnProductListParams,
```

Comment:
If all properties on this object are optional, then "params" should be optional too. Same for any other calls with fully optional params.
```suggestion
    params?: SimpleEarnProductListParams,
```

The SDK will automatically see that this is undefined and will make the request without any params (and it looks cleaner in code).